### ISSUE TITLE
Extract longhandOptmizer to its own postcss module and expand it

### ISSUE THREAD

**Issue by bfred-it** [ben-eb/cssnano#44]
This looks like it could be its own module and perhaps it could be expanded to support more:
https://github.com/ben-eb/cssnano/blob/master/lib/longhandOptimiser.js

Related packages:
https://www.npmjs.com/package/css-shorthand-properties
https://www.npmjs.com/package/shortcss
https://www.npmjs.com/package/shrthnd
https://www.npmjs.com/package/css-shorthand-expand

**Comment by ben-eb** [comment]
Indeed, the functionality here is fairly similar to https://github.com/ben-eb/cssnano/blob/master/lib/borderOptimiser.js - the two form a basis for declaration merging which can be expanded upon. Been meaning to get to this for some time now.

Do you think such a module should also merge these types of declarations?

```css
h1 {
  color: red;
  color: white;
  color: rgba(255, 255, 255, .5);
}

/* output */

h1 {
  color: white;
  color: rgba(255, 255, 255, .5);
}
```

**Comment by bfred-it** [comment]
It looks like duplicate property merging is already taken care by this: https://github.com/ben-eb/postcss-discard-duplicates#input

A postcss-shorthand-merger plugin would only take care of merging these properties https://github.com/gilmoreorless/css-shorthand-properties/blob/master/index.js#L16-L62, if we have all of them of if the user already starts with a shorthand, like:

```css
background: url(image.jpg);
background-size: cover;
```

```
background: url(image.jpg) center / cover;
```

**Comment by ben-eb** [comment]
Duplicate property merging is handled only in the case when the values are the same, because you could have multiple values for `display` (such as flexbox). If you handle it too naively, then that example I gave could be 'optimised' to this:

```css
h1 {
  color: rgba(255, 255, 255, .5);
}
```

Which discards the fallback declaration for older browsers.

**Comment by bfred-it** [comment]
Right, but I don't see why it would be part of a plugin that takes care of merging longhand properties. The only time where it should do it in case where it's merging longhand properties, only as a consequence of merging. 

```
h1 {
  background: url(image.png) top;
  background-position: center;
}
```

**Comment by ben-eb** [comment]
Yeah, two separate modules seems about right.

**Comment by ben-eb** [comment]
Released as `2.3.0`.

We can continue discussing other optimisations on the new tracker.

https://github.com/ben-eb/postcss-merge-longhand
